test/tests/jobs/CreateJobTest.cpp

```diff
@@ -234,4 +240,167 @@ struct CreateJobTest : tpunit::TestFixture {
         command["parentJobID"] = childID;
         tester->executeWaitVerifyContent(command, "405 Cannot create grandchildren");
     }
+
+    void retryRecurringJobs() {
+        SData command("CreateJob");
+        command["name"] = "test";
+        command["repeat"] = "SCHEDULED, +1 HOUR";
+        command["retryAfter"] = "10";
+        tester->executeWaitVerifyContent(command, "402 Recurring auto-retrying jobs are not supported");
+    }
+
+    void retryWithMalformedValue() {
+        SData command("CreateJob");
+        command["name"] = "test";
+        command["retryAfter"] = "10";
+        tester->executeWaitVerifyContent(command, "402 Malformed retryAfter");
+    }
+
+    void retryUnique() {
+        SData command("CreateJob");
+        command["name"] = "test";
+        command["retryAfter"] = "+10 HOUR";
+        command["unique"] = "true";
+        tester->executeWaitVerifyContent(command, "405 Unique jobs can't be retried");
+    }
+
+    void retryLifecycle() {
+        // Create a retryable job
+        SData command("CreateJob");
+        string jobName = "testRetryable";
+        string retryValue = "+1 SECOND";
+        command["name"] = jobName;
+        command["retryAfter"] = retryValue;
+
+        STable response = tester->executeWaitVerifyContentTable(command);
+        string jobID = response["jobID"];
+
+        // Query the db to confirm it was created correctly
+        SQResult originalJob;
+        tester->readDB("SELECT created, jobID, state, name, nextRun, lastRun, repeat, data, priority, parentJobID, retryAfter FROM jobs WHERE jobID = " + jobID + ";", originalJob);
+        ASSERT_EQUAL(originalJob[0][1], jobID);
+        ASSERT_EQUAL(originalJob[0][2], "QUEUED");
+        ASSERT_EQUAL(originalJob[0][3], jobName);
+        ASSERT_EQUAL(originalJob[0][4], originalJob[0][0]);
+        ASSERT_EQUAL(originalJob[0][5], "");
+        ASSERT_EQUAL(originalJob[0][6], "");
+        ASSERT_EQUAL(originalJob[0][7], "{}");
+        ASSERT_EQUAL(SToInt(originalJob[0][8]), 500);
+        ASSERT_EQUAL(SToInt(originalJob[0][9]), 0);
+        ASSERT_EQUAL(originalJob[0][10], retryValue);
+
+        // Get the job
+        command.clear();
+        command.methodLine = "GetJob";
+        command["name"] = jobName;
+        response = tester->executeWaitVerifyContentTable(command);
+
+        ASSERT_EQUAL(response["data"], "{}");
+        ASSERT_EQUAL(response["jobID"], jobID);
+        ASSERT_EQUAL(response["name"], jobName);
+
+        // Query the db and confirm that state, nextRun and lastRun are correct
+        SQResult jobData;
+        tester->readDB("SELECT state, nextRun, lastRun FROM jobs WHERE jobID = " + jobID + ";", jobData);
+
+        ASSERT_EQUAL(jobData[0][0], "RUNQUEUED");
+        string nextRun = jobData[0][1];
+        string lastRun = jobData[0][2];
+        struct tm tm1;
```

Comment:
Can you please explain this in a comment, specifically this `ASSERT_EQUAL(difftime(nextRunTime, lastRunTime), 1);`?